client/src/components/MapView.vue

```diff
@@ -70,9 +74,11 @@ export default defineComponent({
 
       this.map?.setLayoutProperty(styleLayerId, 'visibility', visible ? 'visible' : 'none');
 
-      // Update the router params when toggling layers to visible.
-      if (visible && router.currentRoute.value.query.layer != styleLayerId) {
-        router.push({ query: Object.assign({}, router.currentRoute.value.query, { layer: styleLayerId }) });
+      // Update the router params when toggling layers to visible. Do not update for leadServiceLinesByParcelLayer, which is not a visible layer.
+      if (visible &&
```

Comment:
Could you just pull this out to a boolean value to like shouldUpdateRouterParam = .... for readability

Reply:
Done